File: 歷屆考題/101/101C.php
<?php
$Lmax = 0;

function fs_tree($Lary,$Ldeep,$Lnode) {

	for($i=0;$i<count($Lary);$i++){
		if (substr($Lary[$i],-1) == $Lnode) {
			fs_tree($Lary,$Ldeep+1,substr($Lary[$i],0,1));
		}
		if ($Ldeep > $GLOBALS["Lmax"]) $GLOBALS["Lmax"] = $Ldeep;
    }
}

$Lstr = "1,3^2,3^3,5^4,5^5,0^6,5";
$Lary = explode("^",$Lstr);
print_r($Lary);
fs_tree($Lary,1,"0");

echo $Lmax;
$Lstr2 = 'hello';
print(substr($Lstr2,0,1))
?>

File: 歷屆考題/101/101C.py
Lmax = []
def tree(Lary,Ldeep,Lnode):
    global Lmax
    for i in range(len(Lary)):
        # print(Lary[i][-1])
        if Lary[i][-1] == Lnode:
            # print('tree:',Lary,Lary[i][0])
            tree(Lary,Ldeep+1,Lary[i][0])
        Lmax.append(Ldeep)

Lary = []
for i in range(int(input())-1):
    Lary.append(input())
tree(Lary,1,'0')
print(max(Lmax))